src/main/java/seedu/address/model/event/UniqueEventList.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * A list of Events that ensures that all events presents in its internal data structure are unique. Also,
+ * no null are allowed.
+ */
+public class UniqueEventList implements Iterable<Event> {
+
+    private final ObservableList<Event> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Event> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the events to checks in already present in its internal list.
```

Comment:
Spelling mistakes here? Sentence incoherent

Reply:
oh yes, couple of mistakes in the sentence, thanks for pointing out